python2/httplib2/__init__.py

```diff
@@ -962,7 +977,8 @@ def connect(self):
                 continue
             break
         if not self.sock:
-            raise socket.error(msg)
+            raise socket.error, msg
```

Comment:
This seems like a step backwards to syntax that is invalid for py3? I recognize that we're in the python2 tree here, but it's not clear why this change was made?

Reply:
Yes, I am guilty of surprise here, should've left a comment or put it in separate commit.

I didn't dig deep, so don't fully understand how it works, but having it raise `socket.error()` instance makes regression to tests that fail socket connect. My guess it's related to shadowing `msg` in earlier try/except branch, but that's only a guess, possibly wrong one.

See https://github.com/httplib2/httplib2/pull/81